restli-client/src/main/java/com/linkedin/restli/client/FluentClient.java

```diff
@@ -0,0 +1,38 @@
+package com.linkedin.restli.client;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * A common interface for client that implements FluentAPIs
+ */
+interface FluentClient
+{
+
+  ThreadLocal<List<ExecutionGroup>> _executionGroup = new ThreadLocal<List<ExecutionGroup>>()
```

Comment:
Why do you need a list of execution groups? is this for the lambda within lambda usecase? 
Please add a comment to explain that

Reply:
I actually thought about possible extension with interacting eg from previous context. Then a "stack" alike data structure is needed.